File: Dumper.java
import java.io.File;
import java.io.FileWriter;
import java.lang.reflect.Method;
import java.util.List;
import java.util.Map;

public class Dumper {
    public static void main(String[] args) throws Exception {
        Class<?> wdClass = Class.forName("com.example.data.WorshipData");
        Object instance = wdClass.getField("INSTANCE").get(null);
        
        FileWriter out = new FileWriter("app/src/main/java/com/example/data/WorshipData.kt");
        out.write("package com.example.data\n\n");
        out.write("import java.time.DayOfWeek\n\n");
        out.write("object WorshipData {\n");
        
        // dailyPrayers
        out.write("    val dailyPrayers = mapOf(\n");
        Map<?, ?> daily = (Map<?, ?>) wdClass.getMethod("getDailyPrayers").invoke(instance);
        for (Map.Entry<?, ?> entry : daily.entrySet()) {
            Object pair = entry.getValue();
            String first = (String) pair.getClass().getMethod("getFirst").invoke(pair);
            String second = (String) pair.getClass().getMethod("getSecond").invoke(pair);
            out.write("        DayOfWeek." + entry.getKey().toString() + " to Pair(\"\"\"" + first + "\"\"\", \"\"\"" + second + "\"\"\"),\n");
        }
        out.write("    )\n");
        
        String[] lists = {"Taqeebat", "Munajat", "Tasbeehat", "AzkarList", "Ziyarats", "ZiyaratsOfDays", "SahifaSajjadiya", "GeneralPrayers", "SalawatsOnHujaj", "AhlAlBaytPrayers", "ConcisePrayers"};
        for (String name : lists) {
            String lowerName = name.substring(0, 1).toLowerCase() + name.substring(1);
            if (name.equals("SahifaSajjadiya")) {
                // monthlyDeeds is before SahifaSajjadiya
                out.write("    val monthlyDeeds = mapOf(\n");
                Map<?, ?> monthly = (Map<?, ?>) wdClass.getMethod("getMonthlyDeeds").invoke(instance);
                for (Map.Entry<?, ?> entry : monthly.entrySet()) {
                    out.write("        \"\"\"" + entry.getKey() + "\"\"\" to listOf(\n");
                    List<?> list = (List<?>) entry.getValue();
                    for (Object pair : list) {
                        String first = (String) pair.getClass().getMethod("getFirst").invoke(pair);
                        String second = (String) pair.getClass().getMethod("getSecond").invoke(pair);
                        out.write("            Pair(\"\"\"" + first + "\"\"\", \"\"\"" + second + "\"\"\"),\n");
                    }
                    out.write("        ),\n");
                }
                out.write("    )\n");
            }
            
            out.write("    val " + lowerName + " = listOf(\n");
            List<?> list = (List<?>) wdClass.getMethod("get" + name).invoke(instance);
            for (Object pair : list) {
                String first = (String) pair.getClass().getMethod("getFirst").invoke(pair);
                String second = (String) pair.getClass().getMethod("getSecond").invoke(pair);
                out.write("        Pair(\"\"\"" + first + "\"\"\", \"\"\"" + second + "\"\"\"),\n");
            }
            out.write("    )\n");
        }
        
        out.write("}\n");
        out.close();
        System.out.println("Dumped to app/src/main/java/com/example/data/WorshipData.kt successfully!");
    }
}

File: dumper.kt
import com.example.data.WorshipData
import java.io.File
import java.time.DayOfWeek

fun main() {
    val data = WorshipData.INSTANCE
    val sb = java.lang.StringBuilder()
    sb.append("package com.example.data\n\n")
    sb.append("import java.time.DayOfWeek\n\n")
    sb.append("object WorshipData {\n")
    
    fun escapeString(s: String): String {
        return "\"\"\"" + s + "\"\"\""
    }
    
    sb.append("    val dailyPrayers = mapOf(\n")
    data.dailyPrayers.forEach { (k, v) ->
        sb.append("        DayOfWeek.${k.name} to Pair(${escapeString(v.first)}, ${escapeString(v.second)}),\n")
    }
    sb.append("    )\n")
    
    fun dumpList(name: String, list: List<Pair<String, String>>) {
        sb.append("    val $name = listOf(\n")
        list.forEach { v ->
            sb.append("        Pair(${escapeString(v.first)}, ${escapeString(v.second)}),\n")
        }
        sb.append("    )\n")
    }
    
    dumpList("taqeebat", data.taqeebat)
    dumpList("munajat", data.munajat)
    dumpList("tasbeehat", data.tasbeehat)
    dumpList("azkarList", data.azkarList)
    dumpList("ziyarats", data.ziyarats)
    dumpList("ziyaratsOfDays", data.ziyaratsOfDays)
    
    sb.append("    val monthlyDeeds = mapOf(\n")
    data.monthlyDeeds.forEach { (k, vList) ->
        sb.append("        ${escapeString(k)} to listOf(\n")
        vList.forEach { v ->
            sb.append("            Pair(${escapeString(v.first)}, ${escapeString(v.second)}),\n")
        }
        sb.append("        ),\n")
    }
    sb.append("    )\n")
    
    dumpList("sahifaSajjadiya", data.sahifaSajjadiya)
    dumpList("generalPrayers", data.generalPrayers)
    dumpList("salawatsOnHujaj", data.salawatsOnHujaj)
    dumpList("ahlAlBaytPrayers", data.ahlAlBaytPrayers)
    dumpList("concisePrayers", data.concisePrayers)
    
    sb.append("}\n")
    
    File("app/src/main/java/com/example/data/WorshipData.kt").writeText(sb.toString())
}
